src/Npgsql/ConnectorPool.cs

```diff
@@ -543,7 +490,7 @@ static void PruneIdleConnectors(object state)
 
             for (var i = 0; i < idle.Length; i++)
             {
-                if (pool.State.Total <= pool._min)
+                if (pool.State.Copy().Total <= pool._min)
```

Comment:
I'm a bit lost here, why do we need this copy?

Reply:
The copy does the volatile read, just making sure the read barrier is there. 

Honestly this was part of a search and replace to have each State load be done via Volatile.Read.

Reply:
At the risk of further complicating things: 

Do you think there's any value to reworking how `PoolState` fields are exposed? Given that we (almost?) always want read barriers, should we add some properties to model that, separate from access to the raw fields?

Reply:
That may be a nice cleanup!
Let me commit that real quick